File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        // Position Placement solution 
        int n = nums.size(); 

        while  (nums[0] != nums[nums[0]]){
            swap(nums[0], nums[nums[0]]);
        }

        return nums[0];
    }
};

File: 287-find-the-duplicate-number/find-the-duplicate-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function(nums) {
    let n = nums.length; 
    // sorting approach will work 
     nums.sort( (a,b)=> a-b );
     for (let i=0 ;i<n-1 ;i++ ){
        if (nums[i]== nums[i+1])return nums[i];
     }
     return -1 ;
};